### ISSUE TITLE
Step Over In Debug not working with GO 1.8

### ISSUE THREAD

**Issue by jli3388** [go-lang-plugin-org/go-lang-idea-plugin#2872]
- Plugin version (or commit hash):

- IntelliJ IDEA 2016.3.4
Build #IU-163.12024.16, built on January 31, 2017
JRE: 1.8.0_112-release-408-b6 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
- MacOS Sierra 10.12.3

- What are you trying to do?
   Stepping Over a break point set in code. 

- What would you expect to happen? 
   The debugger stop correctly at the line with the breakpoint, however stepping over (F8) is not working and debugger never auto move to the next line of code and call stack and variable window shows nothing as if the process is still running. 

- What happens?
   after F8, execution seems to be halted and the cursor never move to stop at the next line of code.

**Comment by mnelson-bah** [comment]
May need to upgrade delve to 0.12.2?  https://github.com/derekparker/delve/issues/761

**Comment by winworm** [comment]
the same problem

**Comment by mnelson-bah** [comment]
Workaround:  Install the latest delve e.g. brew install delve, then in IntelliJ:
Help → Edit Custom Properties...
Add the following as a new line
dlv.path=/usr/local/bin/dlv
Save, close and restart the IDE

**Comment by dlsniper** [comment]
Please try the official JetBrains plugin for Go https://plugins.jetbrains.com/plugin/9568-go which has a lot more features than this plugin. As of a few months ago this plugin is not actively maintained anymore and I doubt I'll have time to work on it in the future as well.

**Comment by dlsniper** [comment]
There's also a dedicated IDE if you want to use instead: https://www.jetbrains.com/go/

**Comment by winworm** [comment]
@dlsniper  i try to install  https://plugins.jetbrains.com/plugin/9568-go  ,download the zip and use the method "install plugin from disk"  . but  error  : pligin 'Go' is incompatible with this installation.

i  use the idea of version 2016.3.6

**Comment by dlsniper** [comment]
The plugin should be installed from the IDE. As for supported versions, I think only 2017.1+ are supported, but I can't remember now. Have a look at Gogland as an alternative.

**Comment by winworm** [comment]
@dlsniper  thanks , try the Goglang  maybe the best way.

**Comment by drgorb** [comment]
I just downloaded the latest GoGland version and it stills hand when stepping. It also hangs when trying to evaluate an expression and take an eternity when trying to read the content of variables

**Comment by dlsniper** [comment]
@drgorb please open an issue on https://youtrack.jetbrains.com/issues/Go and describe what you did in order to have this happening. As you described I cannot reproduce it since there's not information to even begin and try to replicate it. Thank you.

**Comment by helloNice** [comment]
> Workaround: Install the latest delve e.g. brew install delve, then in IntelliJ:
> Help → Edit Custom Properties...
> Add the following as a new line
> dlv.path=/usr/local/bin/dlv
> Save, close and restart the IDE

yes.i had the same question, and I used this way, now I'm flying!!!

**Comment by miracle2017** [comment]
> 

that way is still work for me at 2023.7.  strongly recommend you to try this one, because it is easy and quick to try out. it took me 1 minute to solve the problem. unbelievable!

my go version 1.9 and Golang 2021.2.1

**Comment by miracle2017** [comment]
> Workaround: Install the latest delve e.g. brew install delve, then in IntelliJ: Help → Edit Custom Properties... Add the following as a new line dlv.path=/usr/local/bin/dlv Save, close and restart the IDE

that way is still work for me at 2023.7. **strongly recommend you to try this one**, because it is easy and quick to try out. it took me 1 minute to solve the problem. unbelievable!

my go version 1.9 and Golang 2021.2.1

**Comment by binary111** [comment]
> Workaround: Install the latest delve e.g. brew install delve, then in IntelliJ: Help → Edit Custom Properties... Add the following as a new line dlv.path=/usr/local/bin/dlv Save, close and restart the IDE

For Mac os M1 the directory path is as ` /opt/homebrew/Cellar/delve/1.21.0`
